### PR TITLE
[YEET] Add quick helm rollback shared workflow

### PR DESCRIPTION
**Pull request by fountain-draloff**
This workflow takes a tenant matrix and issues a `helm rollback <project-name> 0` for each namespace indicated.

It waits for the rollback to be complete thanks to the `--wait` flag.

This job has very few guard rails, so it should not be used if there's any doubt as to the state of deploys across all tenants given.

### CODE REVIEW

File: .github/workflows/actions/helm-rollback/action.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    - name: 'Rollback Dry Run'
+      if: ${{ inputs.dry-run }}
+      shell: bash
+      run: |
+        echo helm rollback ${{ inputs.project-name }} 0 \
+          -n ${{ inputs.namespace }} \
+          --atomic \
+          --wait \
+          --timeout 60m0s;
+        helm -n ${{ inputs.namespace }} history ${{ inputs.project-name }} --max 2;
```
Comment (mark-davenport-fountain) [inline .github/workflows/actions/helm-rollback/action.yaml:87]: why not use the `--dry-run` command to simulate the rollback.

File: .github/workflows/actions/helm-rollback/action.yaml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    - name: 'Rollback'
+      if: ${{ inputs.dry-run != true }}
+      shell: bash
+      run: |
+        helm rollback ${{ inputs.project-name }} 0 \
+          -n ${{ inputs.namespace }} \
+          --atomic \
+          --wait \
+          --timeout 60m0s;
```
Comment (mark-davenport-fountain) [inline .github/workflows/actions/helm-rollback/action.yaml:97]: This is....long. Might be something we address later, but I think we shouldn't be running into this especially on rollback. 
The clearest place we hit it today is on deploy with long running migrations.

File: .github/workflows/actions/helm-rollback/action.yaml (hunk 1 above)
Comment (fountain-draloff) [reply .github/workflows/actions/helm-rollback/action.yaml:87]: Oh doy, where was my brain

File: .github/workflows/actions/helm-rollback/action.yaml (hunk 2 above)
Comment (fountain-draloff) [reply .github/workflows/actions/helm-rollback/action.yaml]: Reduced to 10 minutes

**Review by mark-davenport-fountain — APPROVED** [review]

**Pull request merged**